### ISSUE TITLE
Timeout when retrieving relation count in MongoDB

### ISSUE THREAD

**Issue by nikolasburk** [prisma/prisma#9651]
### Bug description

I am getting a timeout error when trying to retrieve the count of a relation in MongoDB:

```
$ yarn ts-node index.ts
yarn run v1.22.5
$ /Users/nikolasburk/prisma/releases/whats-new-in-prisma/v3.2.0-demos/playground/sample_mflix/node_modules/.bin/ts-node index.ts
/Users/nikolasburk/prisma/releases/whats-new-in-prisma/v3.2.0-demos/playground/sample_mflix/node_modules/@prisma/client/runtime/index.js:38739
          throw new import_engine_core3.PrismaClientUnknownRequestError(message, this.prisma._clientVersion);
                ^
PrismaClientUnknownRequestError: 
Invalid `prisma.movie.findMany()` invocation in
/Users/nikolasburk/prisma/releases/whats-new-in-prisma/v3.2.0-demos/playground/sample_mflix/index.ts:6:40

  3 const prisma = new PrismaClient()
  4 
  5 async function main() {
→ 6   const allMovies = await prisma.movie.findMany(
  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: RawError { code: "unknown", message: "Command failed (MaxTimeMSExpired): Error in $cursor stage :: caused by :: operation exceeded time limit)" } })
    at cb (/Users/nikolasburk/prisma/releases/whats-new-in-prisma/v3.2.0-demos/playground/sample_mflix/node_modules/@prisma/client/runtime/index.js:38739:17) {
  clientVersion: '3.2.0'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### How to reproduce

Exactly the same steps as in: https://github.com/prisma/prisma/issues/9650

### Prisma information

This time, the Prisma Client query is the following:

```ts
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()

async function main() {
  const allMovies = await prisma.movie.findMany({
    take: 10,
    include: {
      _count: {
        select: {
          comments: true
        }
      }
    },
  })
  console.log(allMovies)
}

main()
  .catch((e) => {
    throw e
  })
  .finally(async () => {
    await prisma.$disconnect()
  })
```

### Environment & setup

- OS: macOS 10.15.7
- Database: MongoDB
- Node.js version: v16.1.0

### Prisma Version

```
$ npx prisma -v
Environment variables loaded from .env
prisma                  : 3.2.0
@prisma/client          : 3.2.0
Current platform        : darwin
Query Engine (Node-API) : libquery-engine afdab2f10860244038c4e32458134112852d4dad (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli afdab2f10860244038c4e32458134112852d4dad (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core afdab2f10860244038c4e32458134112852d4dad (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt afdab2f10860244038c4e32458134112852d4dad (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : afdab2f10860244038c4e32458134112852d4dad
Studio                  : 0.435.0
Preview Features        : mongoDb
```

**Comment by janpio** [comment]
Possibly related: https://github.com/prisma/prisma/issues/8810

**Comment by janpio** [comment]
I think this is fixed now:

1. `mkdir 9651 && cd 9651 && npm init -y && npm install prisma@dev && npx prisma init --datasource-provider mongodb --url ...` with URL from https://github.com/prisma/introspection-ci/blob/f24f50841a84c1b0249f5d63f428bfb089a8ad9a/databases/connections/mongodb_public.txt#L4
2. `npx prisma db pull`
3. Adapt schema to be:
	  ```prisma
	  model Comment {
	    id       String   @id @default(dbgenerated()) @map("_id") @db.ObjectId
	    date     DateTime @db.Date
	    email    String
	    movie_id String   @db.ObjectId
	    movie    Movie   @relation(fields: [movie_id], references: [id])
	    name     String
	    text     String
	  
	    @@map("comments")
	  }
	  
	  model Movie {
	    id                 String    @id @default(dbgenerated()) @map("_id") @db.ObjectId
	    awards             Json
	    cast               String[]
	    countries          String[]
	    directors          String[]
	    fullplot           String?
	    genres             String[]
	    imdb               Json
	    languages          String[]
	    lastupdated        String
	    metacritic         Int?
	    num_mflix_comments Int
	    plot               String?
	    poster             String?
	    rated              String?
	    released           DateTime? @db.Date
	    runtime            Int?
	    title              String
	    tomatoes           Json?
	    type               String
	    writers            String[]
	    /// String: 0.1%, Int32: 99.9%
	    year               Int
	    comments           Comment[]
	  
	    @@map("movies")
	  }
	  
	  model Session {
	    id      String @id @default(dbgenerated()) @map("_id") @db.ObjectId
	    jwt     String
	    user_id String @unique(map: "user_id_1")
	  
	    @@map("sessions")
	  }
	  
	  model Theater {
	    id        String @id @default(dbgenerated()) @map("_id") @db.ObjectId
	    location  Json
	    theaterId Int
	  
	    @@map("theaters")
	  }
	  
	  model User {
	    id          String @id @default(dbgenerated()) @map("_id") @db.ObjectId
	    email       String @unique(map: "email_1")
	    name        String
	    password    String
	    preferences Json?
	  
	    @@map("users")
	  }
	  ```
4. Create `script.js` with this content:
	  ```
	  const { PrismaClient } = require('@prisma/client')
	  
	  const prisma = new PrismaClient()
	  
	  async function main() {
	    const allMovies = await prisma.movie.findMany({
	      take: 10,
	      include: {
	        _count: {
	          select: {
	            comments: true
	          }
	        }
	      },
	    })
	    console.log(allMovies)
	  }
	  
	  main()
	    .catch((e) => {
	      throw e
	    })
	    .finally(async () => {
	      await prisma.$disconnect()
	    })
	  ```

Leads to this output:
```
C:\Users\Jan\Documents\throwaway\9651>node script.js      
[
  {
    id: '573a1390f29313caabcd4135',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [ 'Charles Kayser', 'John Ott' ],
    countries: [ 'USA' ],
    directors: [ 'William K.L. Dickson' ],
    fullplot: 'A stationary camera looks at a large anvil with a blacksmith behind it and one on either side. The smith in the middle draws a heated metal rod from the fire, places it on the anvil, and all three begin a rhythmic hammering. After several blows, the metal goes back in the fire. One smith pulls out a bottle of beer, and they each take a swig. Then, out comes the glowing metal and the hammering resumes.',
    genres: [ 'Short' ],
    imdb: { rating: 6.2, votes: 1189, id: 5 },
    languages: [],
    lastupdated: '2015-08-26 00:03:50.133000000',
    metacritic: null,
    num_mflix_comments: 0,
    plot: 'Three men hammer on an anvil and pass a bottle of beer around.',
    poster: null,
    rated: 'UNRATED',
    released: 1893-05-09T00:00:00.000Z,
    runtime: 1,
    title: 'Blacksmith Scene',
    tomatoes: { viewer: [Object], lastUpdated: [Object] },
    type: 'movie',
    writers: [],
    year: 1893,
    _count: { comments: 0 }
  },
  {
    id: '573a1390f29313caabcd42e8',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [
      'A.C. Abadie',
      "Gilbert M. 'Broncho Billy' Anderson",
      'George Barnes',
      'Justus D. Barnes'
    ],
    countries: [ 'USA' ],
    directors: [ 'Edwin S. Porter' ],
    fullplot: "Among the earliest existing films in American cinema - notable as the first film that presented a narrative story to tell - it depicts a group of cowboy outlaws who hold up a train and rob the passengers. They are then pursued by a Sheriff's posse. Several scenes have color included - all hand tinted.",
    genres: [ 'Short', 'Western' ],
    imdb: { rating: 7.4, votes: 9847, id: 439 },
    languages: [ 'English' ],
    lastupdated: '2015-08-13 00:27:59.177000000',
    metacritic: null,
    num_mflix_comments: 0,
    plot: 'A group of bandits stage a brazen train hold-up, only to find a determined posse hot on their heels.',
    poster: 'https://m.media-amazon.com/images/M/MV5BMTU3NjE5NzYtYTYyNS00MDVmLWIwYjgtMmYwYWIxZDYyNzU2XkEyXkFqcGdeQXVyNzQzNzQxNzI@._V1_SY1000_SX677_AL_.jpg',
    rated: 'TV-G',
    released: 1903-12-01T00:00:00.000Z,
    runtime: 11,
    title: 'The Great Train Robbery',
    tomatoes: {
      viewer: [Object],
      fresh: 6,
      critic: [Object],
      rotten: 0,
      lastUpdated: [Object]
    },
    type: 'movie',
    writers: [],
    year: 1903,
    _count: { comments: 0 }
  },
  {
    id: '573a1390f29313caabcd4323',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [
      'Martin Fuller',
      'Mrs. William Bechtel',
      'Walter Edwin',
      'Ethel Jewett'
    ],
    countries: [ 'USA' ],
    directors: [ 'Harold M. Shaw' ],
    fullplot: "Thanks to the Fresh Air Fund, a slum child escapes his drunken mother for a day's outing in the country. Upon arriving, he and the other children are told a story about a mythical land of no pain. Rather then return to the slum at day's end, the lad seeks to journey to that beautiful land beyond the sunset.",
    genres: [ 'Short', 'Drama', 'Fantasy' ],
    imdb: { rating: 7.1, votes: 448, id: 488 },
    languages: [ 'English' ],
    lastupdated: '2015-08-29 00:27:45.437000000',
    metacritic: null,
    num_mflix_comments: 1,
    plot: 'A young boy, opressed by his mother, goes on an outing in the country with a social welfare group where he dares to dream of a land where the cares of his ordinary life fade.',
    poster: 'https://m.media-amazon.com/images/M/MV5BMTMzMDcxMjgyNl5BMl5BanBnXkFtZTcwOTgxNjg4Mg@@._V1_SY1000_SX677_AL_.jpg',
    rated: 'UNRATED',
    released: 1912-10-28T00:00:00.000Z,
    runtime: 14,
    title: 'The Land Beyond the Sunset',
    tomatoes: { viewer: [Object], lastUpdated: [Object] },
    type: 'movie',
    writers: [ 'Dorothy G. Shore' ],
    year: 1912,
    _count: { comments: 1 }
  },
  {
    id: '573a1390f29313caabcd446f',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [
      'Frank Powell',
      'Grace Henderson',
      'James Kirkwood',
      'Linda Arvidson'
    ],
    countries: [ 'USA' ],
    directors: [ 'D.W. Griffith' ],
    fullplot: "A greedy tycoon decides, on a whim, to corner the world market in wheat. This doubles the price of bread, forcing the grain's producers into charity lines and further into poverty. The film continues to contrast the ironic differences between the lives of those who work to grow the wheat and the life of the man who dabbles in its sale for profit.",
    genres: [ 'Short', 'Drama' ],
    imdb: { rating: 6.6, votes: 1375, id: 832 },
    languages: [ 'English' ],
    lastupdated: '2015-08-13 00:46:30.660000000',
    metacritic: null,
    num_mflix_comments: 1,
    plot: "A greedy tycoon decides, on a whim, to corner the world market in wheat. This doubles the price of bread, forcing the grain's producers into charity lines and further into poverty. The film...",
    poster: null,
    rated: 'G',
    released: 1909-12-13T00:00:00.000Z,
    runtime: 14,
    title: 'A Corner in Wheat',
    tomatoes: { viewer: [Object], lastUpdated: [Object] },
    type: 'movie',
    writers: [],
    year: 1909,
    _count: { comments: 1 }
  },
  {
    id: '573a1390f29313caabcd4803',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [ 'Winsor McCay' ],
    countries: [ 'USA' ],
    directors: [ 'Winsor McCay', 'J. Stuart Blackton' ],
    fullplot: 'Cartoonist Winsor McCay agrees to create a large set of drawings that will be photographed and made into a motion picture. The job requires plenty of drawing supplies, and the cartoonist must also overcome some mishaps caused by an assistant. Finally, the work is done, and everyone can see the resulting animated picture.',
    genres: [ 'Animation', 'Short', 'Comedy' ],
    imdb: { rating: 7.3, votes: 1034, id: 1737 },
    languages: [ 'English' ],
    lastupdated: '2015-08-29 01:09:03.030000000',
    metacritic: null,
    num_mflix_comments: 0,
    plot: 'Cartoon figures announce, via comic strip balloons, that they will move - and move they do, in a wildly exaggerated style.',
    poster: 'https://m.media-amazon.com/images/M/MV5BYzg2NjNhNTctMjUxMi00ZWU4LWI3ZjYtNTI0NTQxNThjZTk2XkEyXkFqcGdeQXVyNzg5OTk2OA@@._V1_SY1000_SX677_AL_.jpg',
    rated: null,
    released: 1911-04-08T00:00:00.000Z,
    runtime: 7,
    title: 'Winsor McCay, the Famous Cartoonist of the N.Y. Herald and His Moving Comics',
    tomatoes: { viewer: [Object], lastUpdated: [Object] },
    type: 'movie',
    writers: [
      'Winsor McCay (comic strip "Little Nemo in Slumberland")',
      'Winsor McCay (screenplay)'
    ],
    year: 1911,
    _count: { comments: 0 }
  },
  {
    id: '573a1390f29313caabcd4eaf',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [ 'Jane Gail', 'Ethel Grandin', 'William H. Turner', 'Matt Moore' ],
    countries: [ 'USA' ],
    directors: [ 'George Loane Tucker' ],
    fullplot: null,
    genres: [ 'Crime', 'Drama' ],
    imdb: { rating: 6, votes: 371, id: 3471 },
    languages: [ 'English' ],
    lastupdated: '2015-09-15 02:07:14.247000000',
    metacritic: null,
    num_mflix_comments: 1,
    plot: 'A woman, with the aid of her police officer sweetheart, endeavors to uncover the prostitution ring that has kidnapped her sister, and the philanthropist who secretly runs it.',
    poster: 'https://m.media-amazon.com/images/M/MV5BYzk0YWQzMGYtYTM5MC00NjM2LWE5YzYtMjgyNDVhZDg1N2YzXkEyXkFqcGdeQXVyMzE0MjY5ODA@._V1_SY1000_SX677_AL_.jpg',
    rated: 'TV-PG',
    released: 1913-11-24T00:00:00.000Z,
    runtime: 88,
    title: 'Traffic in Souls',
    tomatoes: { viewer: [Object], dvd: [Object], lastUpdated: [Object] },
    type: 'movie',
    writers: [],
    year: 1913,
    _count: { comments: 1 }
  },
  {
    id: '573a1390f29313caabcd50e5',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [ 'Winsor McCay', 'George McManus', 'Roy L. McCardell' ],
    countries: [ 'USA' ],
    directors: [ 'Winsor McCay' ],
    fullplot: 'Winsor Z. McCay bets another cartoonist that he can animate a dinosaur. So he draws a big friendly herbivore called Gertie. Then he get into his own picture. Gertie walks through the picture, eats a tree, meets her creator, and takes him carefully on her back for a ride.',
    genres: [ 'Animation', 'Short', 'Comedy' ],
    imdb: { rating: 7.3, votes: 1837, id: 4008 },
    languages: [ 'English' ],
    lastupdated: '2015-08-18 01:03:15.313000000',
    metacritic: null,
    num_mflix_comments: 0,
    plot: 'The cartoonist, Winsor McCay, brings the Dinosaurus back to life in the figure of his latest creation, Gertie the Dinosaur.',
    poster: 'https://m.media-amazon.com/images/M/MV5BMTQxNzI4ODQ3NF5BMl5BanBnXkFtZTgwNzY5NzMwMjE@._V1_SY1000_SX677_AL_.jpg',
    rated: null,
    released: 1914-09-15T00:00:00.000Z,
    runtime: 12,
    title: 'Gertie the Dinosaur',
    tomatoes: { viewer: [Object], lastUpdated: [Object] },
    type: 'movie',
    writers: [ 'Winsor McCay' ],
    year: 1914,
    _count: { comments: 0 }
  },
  {
    id: '573a1390f29313caabcd516c',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [
      'Stanley Hunt',
      'Sarah Constance Smith Hunt',
      'Mrs. George Walkus',
      "Paddy 'Malid"
    ],
    countries: [ 'USA' ],
    directors: [ 'Edward S. Curtis' ],
    fullplot: null,
    genres: [ 'Drama', 'History' ],
    imdb: { rating: 5.8, votes: 223, id: 4150 },
    languages: [ 'English' ],
    lastupdated: '2015-09-16 12:11:37.770000000',
    metacritic: null,
    num_mflix_comments: 1,
    plot: 'Original advertising for the film describes it as a drama of primitive life on the shores of the North Pacific...',
    poster: 'https://m.media-amazon.com/images/M/MV5BMjE3MjAyNzM5NV5BMl5BanBnXkFtZTgwMjA5OTg5NjE@._V1_SY1000_SX677_AL_.jpg',
    rated: null,
    released: 1914-12-07T00:00:00.000Z,
    runtime: 65,
    title: 'In the Land of the Head Hunters',
    tomatoes: {
      viewer: [Object],
      dvd: [Object],
      website: 'http://www.milestonefilms.com/',
      production: 'World Film Corporation',
      lastUpdated: [Object]
    },
    type: 'movie',
    writers: [ 'Edward S. Curtis (story)' ],
    year: 1914,
    _count: { comments: 1 }
  },
  {
    id: '573a1390f29313caabcd5293',
    awards: { wins: 1, nominations: 0, text: '1 win.' },
    cast: [ 'Pearl White', 'Crane Wilbur', 'Paul Panzer', 'Edward Josè' ],
    countries: [ 'USA' ],
    directors: [ 'Louis J. Gasnier', 'Donald MacKenzie' ],
    fullplot: `Young Pauline is left a lot of money when her wealthy uncle dies. However, her uncle's secretary has been named as her guardian until she marries, at which time she will officially take possession of her inheritance. Meanwhile, her "guardian" and his confederates constantly come up with schemes to get rid of Pauline so that he can get his hands on the money himself.`,
    genres: [ 'Action' ],
    imdb: { rating: 7.6, votes: 744, id: 4465 },
    languages: [ 'English' ],
    lastupdated: '2015-09-12 00:01:18.647000000',
    metacritic: null,
    num_mflix_comments: 0,
    plot: "Young Pauline is left a lot of money when her wealthy uncle dies. However, her uncle's secretary has been named as her guardian until she marries, at which time she will officially take ...",
    poster: 'https://m.media-amazon.com/images/M/MV5BMzgxODk1Mzk2Ml5BMl5BanBnXkFtZTgwMDg0NzkwMjE@._V1_SY1000_SX677_AL_.jpg',
    rated: null,
    released: 1914-03-23T00:00:00.000Z,
    runtime: 199,
    title: 'The Perils of Pauline',
    tomatoes: {
      viewer: [Object],
      production: 'Pathè Frères',
      lastUpdated: [Object]
    },
    type: 'movie',
    writers: [
      'Charles W. Goddard (screenplay)',
      'Basil Dickey (screenplay)',
      'Charles W. Goddard (novel)',
      'George B. Seitz',
      'Bertram Millhauser'
    ],
    year: 1914,
    _count: { comments: 0 }
  },
  {
    id: '573a1390f29313caabcd548c',
    awards: { wins: 2, nominations: 0, text: '2 wins.' },
    cast: [
      'Lillian Gish',
      'Mae Marsh',
      'Henry B. Walthall',
      'Miriam Cooper'
    ],
    countries: [ 'USA' ],
    directors: [ 'D.W. Griffith' ],
    fullplot: "Two brothers, Phil and Ted Stoneman, visit their friends in Piedmont, South Carolina: the family Cameron. This friendship is affected by the Civil War, as the Stonemans and the Camerons must join up opposite armies. The consequences of the War in their lives are shown in connection to major historical events, like the development of the Civil War itself, Lincoln's assassination, and the birth of the Ku Klux Klan.",
    genres: [ 'Drama', 'History', 'Romance' ],
    imdb: { rating: 6.8, votes: 15715, id: 4972 },
    languages: [],
    lastupdated: '2015-09-11 00:32:27.763000000',
    metacritic: null,
    num_mflix_comments: 0,
    plot: "The Civil War divides friends and destroys families, but that's nothing compared to the anarchy in the black-ruled South after the war.",
    poster: 'https://m.media-amazon.com/images/M/MV5BYTM4ZDhiYTQtYzExNC00YjVlLTg2YWYtYTk3NTAzMzcwNTExXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SY1000_SX677_AL_.jpg',
    rated: 'NOT RATED',
    released: 1915-03-03T00:00:00.000Z,
    runtime: 165,
    title: 'The Birth of a Nation',
    tomatoes: {
      viewer: [Object],
      dvd: [Object],
      critic: [Object],
      lastUpdated: [Object],
      consensus: 'Racial depictions aside, The Birth of a Nation is a landmark film whose achievements and pioneering techniques remain fully relevant today.',
      rotten: 0,
      production: 'Gravitas',
      fresh: 38
    },
    type: 'movie',
    writers: [
      'Thomas Dixon Jr. (adapted from his novel: "The Clansman: An Historical Romance of the Ku Klux Klan")',
      'Thomas Dixon Jr. (play)',
      'Thomas Dixon Jr. (novel)',
      'D.W. Griffith',
      'Frank E. Woods'
    ],
    year: 1915,
    _count: { comments: 0 }
  }
]
```

```
C:\Users\Jan\Documents\throwaway\9651>npx prisma -v   
Environment variables loaded from .env
prisma                  : 3.10.0-dev.39
@prisma/client          : 3.10.0-dev.39
Current platform        : windows
Query Engine (Node-API) : libquery-engine 34c0995fcfd0aad58922d72470d74b34b232a879 (at node_modules\@prisma\engines\query_engine-windows.dll.node)      
Migration Engine        : migration-engine-cli 34c0995fcfd0aad58922d72470d74b34b232a879 (at node_modules\@prisma\engines\migration-engine-windows.exe)  
Introspection Engine    : introspection-core 34c0995fcfd0aad58922d72470d74b34b232a879 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 34c0995fcfd0aad58922d72470d74b34b232a879 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 34c0995fcfd0aad58922d72470d74b34b232a879
Studio                  : 0.458.0
Preview Features        : mongoDb
```


Can you confirm @nikolasburk?

**Comment by nikolasburk** [comment]
Thanks, this works for me as well now 👍
